main.tf

```diff
@@ -150,10 +153,11 @@ resource "google_sql_database_instance" "main" {
 
 resource "google_sql_user" "main" {
   project         = var.project_id
-  name            = "${google_service_account.runsa.account_id}@${var.project_id}.iam"
-  type            = "CLOUD_IAM_SERVICE_ACCOUNT"
   instance        = google_sql_database_instance.main.name
   deletion_policy = "ABANDON"
+  name            = var.database_type == "postgres" ? "${google_service_account.runsa.account_id}@${var.project_id}.iam" : "foo"
+  type            = var.database_type == "postgres" ? "CLOUD_IAM_SERVICE_ACCOUNT" : null
+  password        = var.database_type == "mysql" ? "bar" : null
```

Comment:
**Question:** Does MySQL allow us to use `type = CLOUD_IAM_SERVICE_ACCOUNT`? I want to see if we can use the same `google_sql_user` definition for both PostgreSQL and MySQL.

Reply:
I tried it but didn't work for me just right out of the box. Maybe we can take this in a follow up PR if this is not blocking?